engines/private/grammar.y

```diff
@@ -0,0 +1,165 @@
+        code1(funcpush);
+        }
+        | fcall ';'         { $$ = $1; }   
+        | if cond body end {
+         	/* else-less if */
+		($1)[1] = (Inst)$3;	/* thenpart */
+		($1)[3] = (Inst)$4; 
+                }	/* end, if cond fails */
+        | if cond body end ELSETOK body end { 
+                /* if with else */
+		($1)[1] = (Inst)$3;	/* thenpart */
+		($1)[2] = (Inst)$6;	/* elsepart */
+		($1)[3] = (Inst)$7; 
+                }	/* end, if cond fails */
+        ;
+
+body:         statement      { $$ = $1; }
+        | '{' statements '}' { $$ = $2; }
+        ;
+
+end:	  /* nothing */		{ code(STOP); $$ = progp; }
+	;
+
+if: IFTOK { $$ = code(ifcode); code3(STOP, STOP, STOP); }
+        ;
+
+cond: '(' expr ')'	{ code(STOP); $$ = $2; }
+        ;
+
+define:  /* nothing */
+        | NAME ',' RECT '(' NUM ',' NUM ',' NUM ',' NUM ')' ',' define  { 
+          Common::Rect *r = new Common::Rect($5->u.val, $7->u.val, $9->u.val, $11->u.val);
+          assert(r->isValidRect()); 
+          defineSymbol($NAME, r); 
+          }
+        | NAME ',' RECT '(' NUM ',' NUM ',' NUM ',' NUM ')' {
+          Common::Rect *r = new Common::Rect($5->u.val, $7->u.val, $9->u.val, $11->u.val);  
+          defineSymbol($NAME, r); 
+          }
+        | NAME ',' define { defineSymbol($NAME, NULL); }
+        | NAME            { defineSymbol($NAME, NULL); }  
+        ;
+
+fcall:    GOTOTOK '(' NAME ')' {
+                               $$ = progp;
+                               code2(strpush, (Private::Inst) Private::constant(STRING, 0, $NAME));
+                               code2(constpush, (Private::Inst) Private::constant(NUM, 1, NULL));
+                               code2(strpush, (Private::Inst) Private::constant(STRING, 0, "goto")); 
+                               code1(funcpush);
+                               }
+
+        | RECT '(' NUM ',' NUM ',' NUM ',' NUM ')' { $$ = progp; }
+        | NAME '(' startp params ')'  {
+                               $$ = $startp;
+                               code2(constpush, (Private::Inst) constant(NUM, $params, NULL));
+                               code2(strpush, (Private::Inst) constant(STRING, 0, $NAME)); 
+                               code1(funcpush);
+                               }
+        ;
+
+startp: /*nothing*/ { $$ = progp; }
+        ;
+
+params:   /* nothing */     { $$ = 0; }
+        | fcall ',' params  { $$ = $3 + 1; }
+        | expr ',' params   { $$ = $3 + 1; }
+        | expr        { $$ = 1; }
+        | fcall       { $$ = 1; }
+        ;
+
+value:    NULLTOK  { code2(Private::constpush, (Private::Inst) Private::constant(NUM, 0, NULL)); }
+        | FALSETOK { code2(Private::constpush, (Private::Inst) Private::constant(NUM, 0, NULL)); }
+        | TRUETOK  { code2(Private::constpush, (Private::Inst) Private::constant(NUM, 1, NULL)); }
+        |  NUM     { code2(Private::constpush, (Private::Inst)$NUM); }
```

Comment:
Looks like there is an extra space before `NUM`

Reply:
Fixed.